src/conduits/RwaOutputConduit3.sol

```diff
@@ -0,0 +1,380 @@
+     * @dev `msg.sender` must have been `mate`d and `to` must have been `pick`ed.
+     */
+    function push() external isMate {
+        _doPush(dai.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Swaps the specified amount of DAI into GEM through the PSM and push it to the recipient address.
+     * @dev `msg.sender` must have been `mate`d and `to` must have been `pick`ed.
+     * @param wad DAI amount.
+     */
+    function push(uint256 wad) external isMate {
+        _doPush(wad);
+    }
+
+    /**
+     * @notice Flushes out any DAI balance to `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     */
+    function quit() external isMate {
+        _doQuit(dai.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Flushes out the specified amount of DAI to the `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     * @param wad DAI amount.
+     */
+    function quit(uint256 wad) external isMate {
+        _doQuit(wad);
+    }
+
+    /**
+     * @notice Flushes out any GEM balance to `usr` address.
+     * @param usr Destination address
+     * @dev Can be called only by admin.
+     */
+    function quitGem(address usr) external auth {
+        gem.transfer(usr, gem.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Calculate amount of GEM received for swapping DAI through PSM.
+     * @param wad DAI amount.
+     */
+    function daiToGem(uint256 wad) public view returns (uint256) {
+        return mul(wad, WAD) / add(WAD, psm.tout()) / to18ConvertionFactor;
+    }
+
+    /**
+     * @notice Swaps the specified amount of DAI into GEM through the PSM and push it to the recipient address.
+     * @param wad DAI amount.
+     */
+    function _doPush(uint256 wad) internal {
+        require(to != address(0), "RwaOutputConduit3/to-not-picked");
+
+        // We might lose some dust here because of rounding errors. I.e.: USDC has 6 dec and DAI has 18.
+        uint256 gemAmount = daiToGem(wad);
+        require(gemAmount > 0, "RwaOutputConduit3/insufficient-swap-gem-amount");
+
+        uint256 prevGemBalance = gem.balanceOf(address(this));
+
+        psm.buyGem(address(this), gemAmount);
+
+        uint256 gemBalance = gem.balanceOf(address(this));
+        uint256 gemPushAmt = sub(gemBalance, prevGemBalance);
+        address _to = to;
+
+        to = address(0);
+        gem.transfer(_to, gemPushAmt);
+
+        emit Push(_to, gemPushAmt);
+    }
+
+    /**
+     * @notice Flushes out the specified amount of DAI to `quitTo` address.
+     * @param wad The DAI amount.
+     */
+    function _doQuit(uint256 wad) internal {
+        require(quitTo != address(0), "RwaOutputConduit3/invalid-quit-to-address");
+        dai.transfer(quitTo, wad);
+        emit Quit(quitTo, wad);
+    }
+
+    /*//////////////////////////////////
+                    Math
```

Comment:
as discussed I think the pattern of having internal functions right after their wrappers is nicer - https://github.com/makerdao/dss-vest/blob/master/src/DssVest.sol#L232

Reply:
++

Reply:
Honestly it makes absolutely no difference where these functions are declared.
We prefer to keep adjacent code after the main subject of the contract.

Reply:
Same as above, will resolve until we have a more general discussion as to how to treat these contracts.